GenumKit/Stencil/CallNode.swift

```diff
@@ -0,0 +1,96 @@
+//
+// GenumKit
+// Copyright (c) 2016 Olivier Halligon
+// MIT Licence
+//
+
+import Stencil
+
+struct CallableBlock: CustomStringConvertible {
+  let parameters: [String]
+  let nodes: [NodeType]
+
+  func context(_ context: Context, arguments: [Variable]) throws -> [String: Any] {
+    var result = [String: Any]()
+
+    for (parameter, argument) in zip(parameters, arguments) {
+      result[parameter] = try argument.resolve(context)
+    }
+
+    return result
+  }
+
+  public var description: String {
+    fatalError("`func` blocks must be called using the `call` tag")
```

Comment:
We should find a way to throw properly instead of crashing with `fatalError` (maybe that can be done by implementing `render` rather than `description` or something?)